src/main/java/core/basesyntax/ColorSupplier.java

```diff
@@ -0,0 +1,23 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class ColorSupplier {
+    private static final String DEFAULT_COLOR = "WHITE";
+    private Random random = new Random();
+    private Color[] colors = Color.values();
+    private String randomColor;
+
+    private void setRandomColor() {
+        this.randomColor = colors[random.nextInt(colors.length)].toString();
+    }
+
+    public String getRandomColor() {
+        setRandomColor();
+        return randomColor;
+    }
```

Comment:
1. Class variables are required only when we cannot use isolated option within method
2. Emun string representation is better to obtain with name() method:
```suggestion
public class ColorSupplier {
    private static final String DEFAULT_COLOR = "WHITE";
    private Random random = new Random();
    private Color[] colors = Color.values();

    public String getRandomColor() {
        return colors[random.nextInt(colors.length)].name();
    }
```